### PR TITLE
feat: use workmail for rootmail feature

### PR DESCRIPTION
**Pull request by megahf**

### CODE REVIEW

File: cdk/src/functions/workmail-organization.is-complete-handler.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  UpdateDefaultMailDomainCommand,
+} from '@aws-sdk/client-workmail';
+import { PutParameterCommand, SSMClient } from '@aws-sdk/client-ssm';
+export const PROP_DOMAIN = 'Domain';
+export const PROP_PARAM_NAME = 'PropagationParamName';
+
+const workmail = new WorkMailClient({ region: 'eu-west-1' });
+const SSM = new SSMClient();
+
+export interface isCompleteEvent extends AWSLambda.CloudFormationCustomResourceEventCommon {
```
Comment (peermuellerxw) [inline cdk/src/functions/workmail-organization.is-complete-handler.ts:15]: I don't get why you use a custom interface here. 
PhysicalResourceId apparently only exists on update and delete events but not on create event. I think a better way would be to use `CloudFormationCustomResourceEvent` as input to the function and then casting to the right type after checking which event was sent.

File: cdk/src/functions/workmail-organization.on-event-handler.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    new GetMailDomainCommand({
+      OrganizationId: workmailOrgId,
+      DomainName: domain,
+    }),
+  );
+
+  // Parse the records for Route53 delete command input
+  const changeRecords: Change[] = [];
+
+  workmail_domain.Records!.forEach((record) => {
```
Comment (peermuellerxw) [inline cdk/src/functions/workmail-organization.on-event-handler.ts:98]: where do you actually set the required DNS records for the create/update case?
You seem to be only deleting them. When I rolled this out in the pipeline account the DNS records were not created and mail got rejected.

**Review by peermuellerxw — CHANGES_REQUESTED** [review]

File: cdk/src/constructs/rootmail-workmail-organization.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import { CustomResource, Duration, Stack, aws_iam as iam, aws_lambda as lambda, aws_ssm as ssm } from 'aws-cdk-lib';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as cr from 'aws-cdk-lib/custom-resources';
+import { Construct, Node } from 'constructs';
+import * as path from 'path';
+import { PROP_DOMAIN, PROP_PARAM_NAME, PROP_HOSTED_ZONE_ID } from '../functions/workmail-organization.on-event-handler';
+
+export interface WorkmailOrganizationProps {
+  readonly domain: string;
+  readonly totalTimeToWireDNS?: Duration;
```
[the remark this answers is not in the document]
Comment (megahf) [reply cdk/src/constructs/rootmail-workmail-organization.ts:10]: No it's not, thanks!

File: cdk/src/functions/workmail-organization.on-event-handler.ts (hunk 2 above)
Comment (megahf) [reply cdk/src/functions/workmail-organization.on-event-handler.ts:118]: The DNS records were automatically created by workmail in the hosted zone when I deploy the stack in standalone-04. I will deploy it again on pipeline02 to see the error logs. If it indeed is the issue, I will create cdk resources for the DNS records.

File: cdk/src/functions/living-docs-dashboard-generator.ts
Comment (peermuellerxw) [inline cdk/src/functions/living-docs-dashboard-generator.ts:1]: These changes are already on main. How did you merge with the main branch?

File: cdk/src/functions/ses-rule-set-function.ts
Comment (peermuellerxw) [inline cdk/src/functions/ses-rule-set-function.ts:1]: Why do we need this function?

File: cdk/src/functions/living-docs-dashboard-generator.ts
Comment (megahf) [reply cdk/src/functions/living-docs-dashboard-generator.ts:1]: I ran "git merge main" on the rootmail-workmail branch

File: cdk/src/constructs/rootmail-workmail-organization.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (peermuellerxw) [reply cdk/src/constructs/rootmail-workmail-organization.ts:10]: Can you remove the parameter if it's not being used?

File: cdk/src/constructs/rootmail-workmail-user.ts
Comment (peermuellerxw) [inline cdk/src/constructs/rootmail-workmail-user.ts:1]: Please remove the file if it's not required

File: cdk/src/functions/workmail-organization.is-complete-handler.ts (hunk 1 above)
Comment (peermuellerxw) [reply cdk/src/functions/workmail-organization.is-complete-handler.ts:14]: Ah ok. Now I understand.
Maybe what you want in this case is CdkCustomResourceIsCompleteEvent:

```
// IsComplete events will contain all normal request fields, as well as those returned from
// the initial onEvent handler.
export type CdkCustomResourceIsCompleteEvent = CdkCustomResourceEvent & CdkCustomResourceResponse;```

File: cdk/src/constructs/rootmail-workmail-user.ts
Comment (megahf) [reply cdk/src/constructs/rootmail-workmail-user.ts:1]: File is needed.

**Pull request closed without merging**